tests/UnifiedSpecTests/UnifiedTestRunner.php

```diff
@@ -357,30 +353,6 @@ private function isClientSideEncryptionSupported(): bool
         return static::isCryptSharedLibAvailable() || static::isMongocryptdAvailable();
     }
 
-    private static function isCryptSharedLibAvailable(): bool
```

Comment:
@eramongodb, @kevinAlbs: The `csfle` [runOnRequirement](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#runonrequirement) dates back to https://github.com/mongodb/specifications/commit/0f65908778188a8d6eeb22571c7791a2b1674234 and [DRIVERS-2017](https://jira.mongodb.org/browse/DRIVERS-2017). Reading the existing language in the unified spec, there's no indication that `crypt_shared` and/or `mongocryptd` are required. 

> If true, the tests MUST only run if the driver and server support Client-Side Field Level Encryption. A server supports CSFLE if it is version 4.2.0 or higher. If false, tests MUST only run if CSFLE is not enabled. If this field is omitted, there is no CSFLE requirement.

The PHPLIB unified test runner currently checks three things:

 * Server FCV >= 4.2
 * PHPC is compiled with libmongocrypt (not just libmongoc)
 * At least one of `crypt_shared` and `mongocryptd` is available

I realize that the only unified CSFLE tests at the moment are for key management, which isn't an enterprise feature. But ignoring that for a moment and assuming legacy tests will eventually be ported to the unified format, was it your intention that the `csfle` requirement also ensure that queryable encryption be supported?

If yes, then I'd like to clarify this in the unified test spec. If no, then I think we should remove the `crypt_shared`/`mongocryptd` check from PHPLIB's unified test runner (and maybe still clarify that in the unified spec).

Reply:
> At least one of `crypt_shared` and `mongocryptd` is available

I expect `crypt_shared` or `mongocryptd` would be required once the unified test format supports automatic encryption. I expect it is not necessary for the current unified test format. I suggest keeping the check for `crypt_shared` or `mongocryptd`, as I expect it will be needed in the future.

> was it your intention that the `csfle` requirement also ensure that queryable encryption be supported?

I think yes. Both "Client-Side Field Level Encryption" and "Queryable Encryption" require libmongocrypt. As of DRIVERS-2454, "In-Use Encryption" can be used to refer to both features.

Reply:
Thanks. I opened https://github.com/mongodb/specifications/pull/1439 and [DRIVERS-2661](https://jira.mongodb.org/browse/DRIVERS-2661) to update the unified/legacy test docs.

Reply:
The `csfle` requirement was added as an (overly?) simple all-or-nothing check for whether anything related to Client Side Encryption is required by the test, which implicitly requires - or at least, was assumed to require - either `mongocryptd` (the only consideration back then) or `crypt_shared` (a new option since the wording was added), since most CSE functions cannot fulfill their specified behavior without them, including the Key management API. I support the clarification of wording that documents the `mongocryptd`/`crypt_shared` requirement; submitted an approving review for https://github.com/mongodb/specifications/pull/1439 accordingly.